packages/nimble-components/src/table/specs/README.md

```diff
@@ -132,13 +132,17 @@ The various APIs/features of the `nimble-table` will be split up amongst several
     -   Define how the action menu gets associated with a particular column
     -   Define the table-level(column-level?) API(s) for applying an action menu to the table (slot, properties, etc...)
 
+_Attributes_
+
+-   `id-field-name` - An optional string attribute that specifies the field name within a row's record to use as a row's ID. If the attribute is not defined or a field with the specified name does not exist in a row's record, a default ID will be created. The default ID is the index of the row's record within the table's `data` property. The value associated with the field name must be unique across all records in the table's data.
```

Comment:
Half formed thought as I leave for the day, feel free to request clarification or defer for later discussion...

Since you mentioned that row IDs are required to build context menus, I'll mention a couple corner cases to consider in case they affect how we want to handle/generate IDs:
1. when the "grouping" feature is enabled, opening an action menu on a "group" row that the table created as a parent of data rows
2. multiselecting rows then opening a context menu

Both of these might be moot if we don't plan to support multi-edit operations via context menus. But if they are supported then we might need to think about APIs that support an array of IDs and/or a way to generate IDs for rows that were created by the table rather than the `data` records.

Reply:
1. At least today, we don't have action menus on a "group" row. If that changes, we'll have to come up with an appropriate way to convey that information to the client. I think we could defer this until the requirement for having group context menus arises.
2. In the SLE grid today, opening the menu will cause only a single row to be selected. I could see that changing in the future, so I was already thinking about proposing that the event that is fired when the menu is opening might contain an array of row IDs. I think we can defer this for the spec about the context menu (likely later this week) when we flesh out exactly what the event details & name will be.

Reply:
Sounds good, thanks for thinking about it.